### PR TITLE
[Improvement] URLs scoped with workspace id

### LINKED ISSUE #5748: [feature] Add workspace-id scope to the protected paths
### More about the feature

By doing this user will be able to open different workspaces on different tabs

### If the feature is approved, would you be willing to submit a PR?
Yes

### PR DESCRIPTION
**Pull request by shah21**
Resolves #5748

Includes deprecating single workspace & new access token flow

### CODE REVIEW

File: frontend/src/_components/PrivateRoute.jsx [hunk 1]
```diff
@@ -1,46 +1,111 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import { Navigate, useLocation } from 'react-router-dom';
 import { authenticationService } from '@/_services';
+import { excludeWorkspaceIdFromURL, appendWorkspaceId } from '../_helpers/utils';
```
Comment (kavinvenkatachalam) [inline frontend/src/_components/PrivateRoute.jsx:4]:
```suggestion
import { excludeWorkspaceIdFromURL, appendWorkspaceId } from '@/_helpers/utils';
```

File: frontend/src/_components/OrganizationManager/List.jsx [hunk 2]
```diff
@@ -1,51 +1,28 @@
 import React, { useState, useEffect } from 'react';
-import { authenticationService, organizationService } from '@/_services';
+import { authenticationService } from '@/_services';
 import { CustomSelect } from './CustomSelect';
+import { getWorkspaceIdFromURL, appendWorkspaceId, getAvatar } from '../../_helpers/utils';
```
Comment (kavinvenkatachalam) [inline frontend/src/_components/OrganizationManager/List.jsx:4]:
```suggestion
import { getWorkspaceIdFromURL, appendWorkspaceId, getAvatar } from '@/_helpers/utils';
```

File: frontend/src/_components/OrganizationManager/CreateOrganization.jsx [hunk 3]
```diff
@@ -1,8 +1,9 @@
 import React, { useState } from 'react';
-import { authenticationService, organizationService } from '@/_services';
+import { organizationService } from '@/_services';
 import AlertDialog from '@/_ui/AlertDialog';
 import { toast } from 'react-hot-toast';
 import { useTranslation } from 'react-i18next';
+import { appendWorkspaceId } from '../../_helpers/utils';
```
Comment (kavinvenkatachalam) [inline frontend/src/_components/OrganizationManager/CreateOrganization.jsx:6]:
```suggestion
import { appendWorkspaceId } from '@/_helpers/utils';
```

File: frontend/src/TooljetDatabase/index.jsx [hunk 4]
```diff
@@ -2,6 +2,7 @@ import React, { createContext, useState, useMemo } from 'react';
 import Layout from '@/_ui/Layout';
 import TooljetDatabasePage from './TooljetDatabasePage';
 import { usePostgrestQueryBuilder } from './usePostgrestQueryBuilder';
+import { authenticationService } from '../_services/authentication.service';
```
Comment (kavinvenkatachalam) [inline frontend/src/TooljetDatabase/index.jsx:5]:
```suggestion
import { authenticationService } from '@/_services/authentication.service';
```

File: frontend/src/HomePage/TemplateLibraryModal/TemplateLibraryModal.jsx [hunk 5]
```diff
@@ -8,6 +8,7 @@ import _ from 'lodash';
 import TemplateDisplay from './TemplateDisplay';
 import { useNavigate } from 'react-router-dom';
 import { useTranslation } from 'react-i18next';
+import { getWorkspaceId } from '../../_helpers/utils';
```
Comment (kavinvenkatachalam) [inline frontend/src/HomePage/TemplateLibraryModal/TemplateLibraryModal.jsx:11]:
```suggestion
import { getWorkspaceId } from '@/_helpers/utils';
```

File: frontend/src/HomePage/SwitchWorkspacePage/index.jsx [hunk 6]
```diff
@@ -0,0 +1,102 @@
+import React, { useState, useEffect } from 'react';
+import { Modal } from 'react-bootstrap';
+import { useTranslation } from 'react-i18next';
+import { getAvatar, appendWorkspaceId } from '../../_helpers/utils';
```
Comment (kavinvenkatachalam) [inline frontend/src/HomePage/SwitchWorkspacePage/index.jsx:4]:
```suggestion
import { getAvatar, appendWorkspaceId } from '@/_helpers/utils';
```

File: frontend/src/HomePage/BlankPage.jsx [hunk 7]
```diff
@@ -4,6 +4,7 @@ import TemplateLibraryModal from './TemplateLibraryModal/';
 import { useTranslation } from 'react-i18next';
 import { libraryAppService } from '@/_services';
 import EmptyIllustration from '@assets/images/no-apps.svg';
+import { getWorkspaceId } from '../_helpers/utils';
```
Comment (kavinvenkatachalam) [inline frontend/src/HomePage/BlankPage.jsx:7]:
```suggestion
import { getWorkspaceId } from '@/_helpers/utils';
```

File: frontend/src/HomePage/AppCard.jsx [hunk 8]
```diff
@@ -9,6 +9,7 @@ import configs from './Configs/AppIcon.json';
 import { Link, useNavigate } from 'react-router-dom';
 import urlJoin from 'url-join';
 import { useTranslation } from 'react-i18next';
+import { getPrivateRoute } from '../_helpers/routes';
```
Comment (kavinvenkatachalam) [inline frontend/src/HomePage/AppCard.jsx:12]:
```suggestion
import { getPrivateRoute } from '@/_helpers/routes';
```

File: frontend/src/GlobalDatasources/index.jsx [hunk 9]
```diff
@@ -3,6 +3,7 @@ import { useNavigate } from 'react-router-dom';
 import Layout from '@/_ui/Layout';
 import { globalDatasourceService } from '@/_services';
 import { GlobalDataSourcesPage } from './GlobalDataSourcesPage';
+import { authenticationService } from '../_services/authentication.service';
```
Comment (kavinvenkatachalam) [inline frontend/src/GlobalDatasources/index.jsx:6]:
```suggestion
import { authenticationService } from '@/_services/authentication.service';
```

File: frontend/src/Editor/QueryManager/QueryEditors/TooljetDatabase/ToolJetDbOperations.jsx [hunk 10]
```diff
@@ -9,13 +9,15 @@ import { toast } from 'react-hot-toast';
 import Select from '@/_ui/Select';
 import { queryManagerSelectComponentStyle } from '@/_ui/Select/styles';
 import { useMounted } from '@/_hooks/use-mount';
+import { authenticationService } from '../../../../_services/authentication.service';
```
Comment (kavinvenkatachalam) [inline frontend/src/Editor/QueryManager/QueryEditors/TooljetDatabase/ToolJetDbOperations.jsx:12]:
```suggestion
import { authenticationService } from '@/_services/authentication.service';
```

File: frontend/src/Editor/ManageAppUsers.jsx [hunk 11]
```diff
@@ -8,11 +8,12 @@ import { debounce } from 'lodash';
 import Textarea from '@/_ui/Textarea';
 import { withTranslation } from 'react-i18next';
 import { Link } from 'react-router-dom';
+import { getPrivateRoute } from '../_helpers/routes';
```
Comment (kavinvenkatachalam) [inline frontend/src/Editor/ManageAppUsers.jsx:11]:
```suggestion
import { getPrivateRoute } from '@/_helpers/routes';
```

File: frontend/src/App/App.jsx [hunk 12]
```diff
@@ -49,23 +57,151 @@ class AppComponent extends React.Component {
...
+              this.authorizeUserAndHandleErrors(current_organization_id);
+            } else {
+              this.updateCurrentSession({
+                current_organization_id,
+              });
+            }
+          })
+          .catch(() => {
+            if (!this.isThisWorkspaceLoginPage(true) && !isApplicationsPath) {
+              // window.location = `${getSubpath() ?? ''}/login`;
```
Comment (kavinvenkatachalam) [inline frontend/src/App/App.jsx:103]:
```suggestion
```

**Review by kavinvenkatachalam — CHANGES_REQUESTED** [review]
@shah21 Requested a minor changes in importing the files to make it consistent and better readability. 

Rest every thing looks good to me

**Review by kavinvenkatachalam — APPROVED** [review]

**Review by gsmithun4 — APPROVED** [review]

**Pull request merged**